support-services/src/main/scala/com/gu/support/catalog/CatalogService.scala

```diff
@@ -14,6 +14,34 @@ object CatalogService {
 }
 
 class CatalogService(val environment: TouchPointEnvironment, jsonProvider: CatalogJsonProvider) extends LazyLogging {
+
+  private def getRatePlanId(billingPeriod: BillingPeriod, fulfilmentOptions: FulfilmentOptions)
+    = getProductRatePlan(environment, billingPeriod,fulfilmentOptions, NoProductOptions).map(_.id).getOrElse("")
+
+  private def fetchQuarterlyPrice(quarterlyId: ProductRatePlanId, sixWeeklyPriceList: Pricelist, catalogPrices: List[Pricelist]) = {
+    Pricelist(sixWeeklyPriceList.productRatePlanId,
+      catalogPrices.find(_.productRatePlanId == quarterlyId).map(_.prices).getOrElse(sixWeeklyPriceList.prices)
+    )
+  }
+
+  def adjustSixWeeklyPriceList(c: Catalog) = {
+    // The price stored for the 6 weekly billing period in Zuora is £6 - the price of the
+    // promotion. It is much more use from the point of view of the site to have the subscription
+    // price, ie. the quarterly price as the £6 is available through the introductory promotion object
+    val ratePlanIdsToSwap = Map(
+      getRatePlanId(SixWeekly, Domestic) -> getRatePlanId(Quarterly, Domestic),
+      getRatePlanId(SixWeekly, RestOfWorld) -> getRatePlanId(Quarterly, RestOfWorld)
+    )
+
+    Catalog(
+      c.prices.map(priceList =>
+        ratePlanIdsToSwap.get(priceList.productRatePlanId)
+          .map(quarterlyId => fetchQuarterlyPrice(quarterlyId, priceList, c.prices))
```

Comment:
You can flatten the lambda if you want: `.map(fetchQuarterlyPrice(_, priceList, c.prices))`